tests/unit_tests/test_dendrite.py

```diff
@@ -334,3 +341,75 @@ async def test_dendrite__call__handles_http_error_response(
 
     assert synapse.axon.status_code == synapse.dendrite.status_code == status_code
     assert synapse.axon.status_message == synapse.dendrite.status_message == message
+
+
+@pytest.mark.parametrize(
+    "exception, expected_status_code, expected_message, synapse_timeout, synapse_ip, synapse_port, request_name",
+    [
+        (
+            aiohttp.ClientConnectorError(Mock(), Mock()),
+            DENDRITE_ERROR_MAPPING[aiohttp.ClientConnectorError][0],
+            f"{DENDRITE_ERROR_MAPPING[aiohttp.ClientConnectorError][1]} at 127.0.0.1:8080/test_request",
+            None,
+            "127.0.0.1",
+            "8080",
+            "test_request_client_connector_error",
+        ),
+        (
+            asyncio.TimeoutError(),
+            DENDRITE_ERROR_MAPPING[asyncio.TimeoutError][0],
+            f"{DENDRITE_ERROR_MAPPING[asyncio.TimeoutError][1]} after 5 seconds",
+            5,
+            None,
+            None,
+            "test_request_timeout",
+        ),
+        (
+            aiohttp.ClientResponseError(Mock(), Mock(), status=404),
+            "404",
+            f"{DENDRITE_ERROR_MAPPING[aiohttp.ClientResponseError][1]}: 404, message=''",
+            None,
+            None,
+            None,
+            "test_request_client_response_error",
+        ),
+        (
+            Exception("Unknown error"),
+            DENDRITE_DEFAULT_ERROR[0],
+            f"{DENDRITE_DEFAULT_ERROR[1]}: Unknown error",
+            None,
+            None,
+            None,
+            "test_request_unknown_error",
+        ),
+    ],
+    ids=[
+        "ClientConnectorError",
+        "TimeoutError",
+        "ClientResponseError",
+        "GenericException",
+    ],
+)
+def test_process_error_message(
+    exception,
+    expected_status_code,
+    expected_message,
+    synapse_timeout,
+    synapse_ip,
+    synapse_port,
+    request_name,
+):
+    # Arrange
+    dendrite = DendriteMixin()
```

Comment:
Are we sure we want to use `DendriteMixin` here rather than the `dendrite` class?

Reply:
Good catch - updating now.